extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/ClientServerSessionCache.java

```diff
@@ -225,6 +221,17 @@ void createSessionRegionOnServers() {
     }
   }
 
+  Region<String, HttpSession> createLocalSessionRegionWithRegisterInterest() {
+    Region<String, HttpSession> region = createLocalSessionRegion();
+
+    // register interest are needed for proxy or caching-proxy client:
+    // to get updates from server if local cache is enabled;
+    // to get callbacks for listener invocation for proxy client.
+    region.registerInterestForAllKeys(InterestResultPolicy.KEYS);
```

Comment:
We could registerInterest in a common place for both user and internally created session regions.